src/core/client_channel/client_channel_filter.cc

```diff
@@ -445,10 +445,9 @@ const grpc_channel_filter ClientChannelFilter::kFilterVtableWithoutPromises = {
 
 namespace {
 
-ClientChannelServiceConfigCallData* GetServiceConfigCallData(
-    grpc_call_context_element* context) {
-  return static_cast<ClientChannelServiceConfigCallData*>(
-      context[GRPC_CONTEXT_SERVICE_CONFIG_CALL_DATA].value);
+ClientChannelServiceConfigCallData* GetServiceConfigCallData(Arena* arena) {
+  return DownCast<ClientChannelServiceConfigCallData*>(
```

Comment:
Can't this just call `arena->GetContext<ClientChannelServiceConfigCallData>()`, because of the `ContextSubclass` template override in client_channel_internal.h?

If so, this `GetServiceConfigCallData()` function is probably no longer needed -- we could just inline the above code in all the places that currently call it.

Reply:
The arena type doesn't know about the casting stuff... and I think I want to hold that separate.

Long term most code should be calling the free function `GetContext<>` directly, and very few places should need to go down the the arena api -- but that long term is "once we ditch filter apis"